worker/uniter/runner/runner.go

```diff
@@ -391,14 +429,20 @@ func (b *bufferAdaptor) Messagef(isPrefix bool, message string, args ...interfac
 	b.outCopy.WriteString(formattedMessage)
 }
 
-func (runner *runner) runCharmHookOnRemote(hookName string, env []string, charmLocation string) error {
+func (runner *runner) runCharmHookOnRemote(hookName string, env []string, charmLocation string) (HookHandlerType, error) {
 	charmDir := runner.paths.GetCharmDir()
 	hook := filepath.Join(charmDir, filepath.Join(charmLocation, hookName))
 
+	// TODO(achilleasa): we need to update the remote executor to recognize
+	// the presence of the hook dispatcher script and report that in its
+	// response payload. As this requires several changes to other packages,
+	// we will assume for now that we are always using the legacy hook scripts.
+	hookHandlerType := LegacyHookHander
```

Comment:
We probably need to implement some mechanism for detecting the hook handler type that was used by the remote before this PR can land.

Reply:
This will be covered in a follow up PR to address k8s actions using dispatch.